sample-apps/auto-instrumentation-java-spark/src/main/java/com/amazon/sampleapp/MetricEmitter.java

```diff
@@ -0,0 +1,232 @@
+  static int currentTimeAlive = 0;
+  static int currentBytesSent = 0;
+  static int currentActiveThreads = 0;
+
+  static String API_COUNTER_METRIC = "totalBytesSent";
+
+  static String API_ASYNC_COUNTER_METRIC = "totalApiRequests";
+
+  static String API_HISTOGRAM_METRIC = "latencyTime";
+
+  static String API_RANDOM_COUNTER_METRIC = "timeAlive";
+
+  static String API_RANDOM_ASYNC_UPDOWN_METRIC = "totalHeapSize";
+
+  static String API_RANDOM_UPDOWN_METRIC = "threadsActive";
+
+  static String API_RANDOM_GAUGE = "cpuUsage";
+
+  // The below API name and status code dimensions are currently shared by all metrics observer in
+  // this class.
+  String apiNameValue = "";
+  String statusCodeValue = "";
+
+  //Declaring variables to track values of metrics
+  DoubleHistogram apiLatencyHistogram;
+  LongCounter bytesCounter;
+  LongCounter timeCounter;
+  LongCounter threadsCounter;
+
+  long totalApiRequests;
+  long totalHeapSize;
+  long cpuUsage;
+
+
+  public MetricEmitter() {
+    Meter meter =
+        GlobalOpenTelemetry.meterBuilder("aws-otel").setInstrumentationVersion("1.0").build();
+
+    // give a instanceId appending to the metricname so that we can check the metric for each round
+    // of integ-test
+    String totalBytesSentName = API_COUNTER_METRIC;
+    String totalApiRequestsName = API_ASYNC_COUNTER_METRIC;
+    String latencyTimeName = API_HISTOGRAM_METRIC;
+    String timeAliveName = API_RANDOM_COUNTER_METRIC;
+    String totalHeapSizeName = API_RANDOM_ASYNC_UPDOWN_METRIC;
+    String threadsActiveName = API_RANDOM_UPDOWN_METRIC;
+    String cpuUsageName = API_RANDOM_GAUGE;
+
+    String instanceId = System.getenv("INSTANCE_ID");
+    if (instanceId != null && !instanceId.trim().equals("")) {
+        totalBytesSentName = API_COUNTER_METRIC + "_" + instanceId;
+        totalApiRequestsName = API_ASYNC_COUNTER_METRIC + "_" + instanceId;
+        latencyTimeName = API_HISTOGRAM_METRIC + "_" + instanceId;
+        timeAliveName = API_RANDOM_COUNTER_METRIC + "_" + instanceId;
+        totalHeapSizeName = API_RANDOM_ASYNC_UPDOWN_METRIC + "_" + instanceId;
+        threadsActiveName = API_RANDOM_UPDOWN_METRIC + "_" + instanceId;
+        cpuUsageName = API_RANDOM_GAUGE + "_" + instanceId;
+    }
+
+    //building synchronous request-based counter metric
+    bytesCounter = meter
+            .counterBuilder(totalBytesSentName)
+            .setDescription("API request load sent in bytes")
+            .setUnit("mb")
+            .build();
+
+    //building asynchronous request-based counter metric
+    meter
+            .counterBuilder(totalApiRequestsName)
+            .setDescription("Total amount of API requests to endpoint")
+            .setUnit("1")
+            .buildWithCallback(measurement ->
+                    measurement.record(
+                            totalApiRequests,
+                            Attributes.of(DIMENSION_API_NAME, apiNameValue, DIMENSION_STATUS_CODE, statusCodeValue)));
```

Comment:
replace these attributes with the ones mentioned, following the requirements doc standard